addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4348,7 +4348,7 @@ registry.BackgroundPosition = SnippetOptionWidget.extend({
      * @private
      * @param {boolean} activate toggle the overlay on (true) or off (false)
      */
-    _toggleBgOverlay: function (activate) {
+    _toggleBgOverlay: async function (activate) {
```

Comment:
In fact there are problems with this:
- This method is now async -> not really stable because custo may not rely on the fact that everything that is done after your `await` is not done immediately anymore
- This method is now async -> all callers should somehow take that into account (here you did not adapt any of them)

Seeing the calls, there is only one which calls this method with a `true` parameter and this method is in an async environment: the easiest and most stable fix is to keep this method here synchrone and rely (as before) on the fact the snippet is in the viewport. Then adapt the only caller to first scroll to the snippet before calling this function.